api4/system.go

```diff
@@ -515,3 +517,38 @@ func getRedirectLocation(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(model.MapToJson(m)))
 	return
 }
+
+func pushNotificationsAck(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
Sorry I don't understand the flow, but where is this ack coming from?

Reply:
This comes from the mobile app, the idea is, this will be implemented in the mobile app and, if the server configuration enables PushNotificationAck, it will send an Ack message as soon as it can. This way we can confirm the entire flow of the push notification.

Reply:
0/5 and may be asking too much even if it makes sense, but... should the DoAck/handleAck be implemented by the notification tracking plugins? I understand that installing the necessary hook in the client may not be easily possible, but wouldn't that be "The Right Thing to Do"?

Reply:
The thing is we can wait until we can create client plugins in the mobile apps, but anyway makes sense to provide the entire push notification flow in this case. Actually this way we can implement other kind of plugins, not only for monitoring. We can implement anonymization of notifications (like, remove any private/sensitive data from the message, and give it to the user using the ACK, avoiding the data to travel through apple and google servers). We can implement "push notification enrichment", like add during the ack attachments to the notification, and show it (the push notification itself have some limitations of size).

Basically, looks like a good thing to have out of the box, without the need of extra plugin for client, and we really need a better way to understand how notifications are behaving right now, and I think we can't wait until mobile plugins are available.